app/dao/fact_notification_status_dao.py

```diff
@@ -86,18 +77,19 @@ def query_for_fact_status_data(table, start_date, end_date, notification_type, s
 
 
 @autocommit
-def update_fact_notification_status(data, process_day, notification_type):
+def update_fact_notification_status(transit_data, process_day, notification_type, service_id):
     table = FactNotificationStatus.__table__
     FactNotificationStatus.query.filter(
         FactNotificationStatus.bst_date == process_day,
-        FactNotificationStatus.notification_type == notification_type
+        FactNotificationStatus.notification_type == notification_type,
+        FactNotificationStatus.service_id == service_id,
     ).delete()
 
-    for row in data:
+    for row in transit_data:
         stmt = insert(table).values(
```

Comment:
Optional: again, I know you didn't name this, but would it be possible to rename `stmt` to `statement`, or even better `new_notification_status` or something like that? It would reduce mental load when reading this part of code.

Reply:
Hmm...I'm not sure we benefit from a variable there at all.

I've made it an inline expression in https://github.com/alphagov/notifications-api/pull/3425/commits/9686595fa85c1fc5458f26c3afcc893a8aa93351 (so no more `stmt` 😄).

Does that work for you?